### ISSUE TITLE
Idea: if mac arm64 build becomes available, download this update on ARM macs

### ISSUE THREAD

**Issue by bartoszhernas** [electron-userland/electron-builder#5672]
macOS universal Electron build are twice as big (and more).
Because of that at FYM.fm we plan to provide separate builds for x64 and ARM64 architecture for macOS.

Currently this is problematic for our current users that are on x64 version of the app, running via Rosetta.
Even if we release the new ARM64 build, the users will be stuck on Intel version unless they download app from scratch from our website.

I checked the code and the fix seems easy, but I wanted to discuss first if this is something that electron-updater would like to do and if I should make a pull request for it.

Basically if I am on ARM Mac, using Intel version, I would like the updater to download ARM version for the update.

**Comment by mmaietta** [comment]
Would using `mac: { defaultArch: 'arm64' }` in your config to get the right filename for it to default updating to arm64 if process.arch === arm64?

**Comment by bartoszhernas** [comment]
Ah, you  are right, the comment mislead me. Indeed, thank you for pointing it out.

**Comment by praneetrohida** [comment]
> Actually, this functionality already exists without any electron-builder config changes.
> ```
> if (files.some(isArm64)) {
>   files = files.filter(file => (process.arch === "arm64") === isArm64(file))
> }
> ```

@mmaietta I'm struggling to see how this will work since apps running in Rosetta will have the emulated process.arch value (x64), so they will get updated to the x64 version and not arm64. Correct me if I'm wrong.

[Source](https://stackoverflow.com/questions/65346260/get-real-architecture-of-m1-mac-regardless-of-rosetta)

**Comment by mmaietta** [comment]
Hmm, great callout.

Hey @davej , you worked on https://github.com/electron-userland/electron-builder/pull/5524. Is your solution working for you? (Allow arm64 macs to update to x64 version if no arm64 version available)

**Comment by praneetrohida** [comment]
Can we reopen this issue?

**Comment by mmaietta** [comment]
@praneetrohida would you be willing to tackle this? 
It's simple to get a local dev env setup with your local project(s):
https://github.com/electron-userland/electron-builder/blob/master/CONTRIBUTING.md#to-setup-a-local-dev-environment

**Comment by praneetrohida** [comment]
@mmaietta I can open a PR but I don't have access to an arm64 mac machine to test it

**Comment by bartoszhernas** [comment]
I will make a PR today (Tallinn time) and I have an ARM Mac :)

**Comment by davej** [comment]
> Hey @davej , you worked on #5524. Is your solution working for you? (Allow arm64 macs to update to x64 version if no arm64 version available)

@mmaietta @praneetrohida I only tested for updating Intel app on M1 to another Intel app on M1. I didn't think of testing the other scenario you mentioned. It makes 100% sense to me though. I suspect that you're correct that it will not allow an emulated Intel version to update to an M1 native app.

**Comment by bartoszhernas** [comment]
Based on this: https://stackoverflow.com/questions/65346260/get-real-architecture-of-m1-mac-regardless-of-rosetta the current implementation won't work as suggested by @praneetrohida 

I will post PR soon
